sos/plugins/kubernetes.py

```diff
@@ -15,40 +15,130 @@
+
+            resources = [
+                'events', 'limitrange', 'pods', 'pvc', 'rc',
+                'resourcequota', 'services'
+            ]
+
+            for res in resources:
+                self.add_cmd_output(
+                    "{} {} {}".format(
+                        kube_bin,
+                        kube_get,
+                        res
+                    )
+                )
+            resources.remove('events')
+            # nodes and pvs are not namespaced, must pull separately
+            self.add_cmd_output([
+                "{} get -o json nodes".format(kube_bin),
+                "{} get -o json pv".format(kube_bin)
+                ]
+            )
+
+            if self.get_option('describe'):
+                resources = resources + ['nodes', 'namespaces']
+                for res in resources:
+                    r = self.get_command_output(
+                        "{} get {}".format(
+                            kube_bin,
+                            res
+                        )
+                    )
+                    if r['status'] == 0:
+                        k_list = [k.split()[0] for k in
+                                  r['output'].splitlines()[1:]
+                                  ]
+                        for k in k_list:
+                            self.add_cmd_output(
+                                "{} describe {} {}".format(
+                                    kube_bin,
+                                    res,
+                                    k
+                                )
+                            )
+
+            if self.get_option('podlogs'):
+                r = self.get_command_output(
+                    "{} get pods".format(kube_bin)
+                )
+                if r['status'] == 0:
+                    pods = [p.split()[0] for p in
+                            r['output'].splitlines()[1:]
+                            ]
+                    for pod in pods:
+                        self.add_cmd_output(
+                            "{} logs {}".format(kube_bin, pod)
+                        )
+
+    def postproc(self):
+        # First, clear sensitive data from the json output collected.
+        # This will mask values when the "name" looks susceptible of
+        # values worth obfuscating, i.e. if the name contains strings
+        # like "pass", "pwd", "key" or "token"
+        env_regexp = r'(?P<var>{\s*"name":\s*[^,]*' \
+                     r'(pass|pwd|key|token|cred)[^,]*,\s*"value":)[^}]*'
+        self.do_cmd_output_sub('kubectl', env_regexp,
+                               r'\g<var> "********"', ignore_case=True)
 
+        # Next, we need to handle the private keys and certs in describe
+        # output that is not hit by the previous iteration.
+        key_regexp = r'-----BEGIN ([A-Z]*).*?-----END ([A-Z].*)-----'
+        self.do_cmd_output_sub('describe', key_regexp,
+                               r'********', dotall=True)
```

Comment:
Needs to be more generic than `do_cmd_key_cert_scrub()` since this could be used to remove other line-delimited chunks in the future. I'm leaning towards `do_cmd_output_splice()` since we're splicing out a defined chunk (but again it's not great because most people think of splicing in rather than out..).

> we could have a dozen or so certs/private keys and ending up with a file littered with

I don't see how that's harmful? It preserves the maximum of the original context and reduces the amount of editing needed if there's ever a need to restore new values (not so likely with command output but this is common with file handling & we should have a matching post hook for those).